### PR TITLE
Transiction refactored agent to asynchronous model

### PR DESCRIPTION
**Pull request by ivar-lazzaro**
- Introduce Port Manager. This module takes care of managing
the async comunication between Neutron and the Opflex Agent
- Switch to the request_* RPC calls.

To avoid increasing the chance of conflicts even further, some
of the methods currently on the main agent class were not moved
to the port manager. This will be done eventually.

### CODE REVIEW

**Comment by ivar-lazzaro** [comment]
Still needs server-side changes, but should be ok to start looking at it

**Comment by ivar-lazzaro** [comment]
Jishnu's testing shows that the agent is working fine, we can start the review-merging process for this.

**Comment by amittbose** [comment]
Looks good, a couple of nits. Will merge once the server-side changes have been committed.

File: opflexagent/utils/port_managers/port_manager_base.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    """ Port Manager base class
+
+    Defines an interface between the GBP opflex Agent and the Neutron server.
+    The port manager takes care of requesting and processing Neutron's ports
+    related info. Uses the gbp_agent for applying ports' configuration.
+    """
+
+    @abc.abstractmethod
+    def initialize(self, host, gbp_agent, config):
+        """ EP Manager initialization method.
```
Comment (amittbose) [inline opflexagent/utils/port_managers/port_manager_base.py]: nit: EP Manager?

File: opflexagent/gbp_ovs_agent.py [hunk 2]
```diff
@@ -565,6 +542,8 @@ def create_agent_config_map(conf):
     agent_config = {}
     agent_config['epg_mapping_dir'] = conf.OPFLEX.epg_mapping_dir
     agent_config['opflex_networks'] = conf.OPFLEX.opflex_networks
+    agent_config['gpb_request_timeout'] = conf.OPFLEX.gpb_request_timeout
```
Comment (amittbose) [inline opflexagent/gbp_ovs_agent.py]: gpb -- is this a typo?

**Pull request merged**